src/niamoto/core/plugins/widgets/bar_plot.py

```
@@ -0,0 +1,327 @@
+                df_plot["_auto_color"] = df_plot[category_field].astype(str)
+                color_field = "_auto_color"
+
+                # Generate colors based on unique categories
+                unique_categories = df_plot[category_field].unique()
+                num_categories = len(unique_categories)
+                generated_colors = generate_colors(num_categories)
+
+                # Create a color mapping
+                color_discrete_map = {
+                    str(cat): color
+                    for cat, color in zip(unique_categories, generated_colors)
```

Ruff (0.11.9)

`zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)